### DISCUSSION TITLE
Put ziggy.js in repo or .gitignore?

### DISCUSSION THREAD

**Discussion by plweil** [tighten/ziggy#506] [Q&A] [answered]
How are people dealing with ziggy and deployments? After integrating ziggy into my dev environment, I'm now preparing to deploy a test branch. Initially I've had the `ziggy.js` file in `.gitignore`, with the idea of generating a new `ziggy.js` on the server during deploy (e.g., by running the artisan task in capistrano). I can see now that this isn't going to work: webpack will spit out errors because it can't resolve the references to `ziggy.js`, which doesn't exist yet.

So the choices seem to be 1) put the generated, full-fledged `ziggy.js` file in the git repo; or 2) generate a blank `ziggy.js` file for the repo, and then run the `artisan ziggy:generate` command during deploy. Are there any real downsides to putting `ziggy.js` in git, even though it feels "wrong"? What are others doing?

**Comment by bakerkretzmar** [comment, marked as answer]
Can you run `php artisan ziggy:generate` before your Webpack build, so the file does exist?

**Reply by plweil** [reply]
Yeah, I think we just need to find where/when to run that artisan task. After composer, but before webpack/Mix.

**Reply by plweil** [reply]
We came up with a solution that runs the task as a composer post-install script. Which works fine for deployments. However...in the dev or local environment, you need to have a ziggy.js file in place before running `npm install`. This can cause an issue whenever you pull a branch to which someone has added/edited node modules. As far as I can see, you can either run the task manually before running `npm install` (which is a little inconvenient), or possibly keep an empty, "placeholder" `ziggy.js` file in the repo. Either option feels a little hackish. I wonder what others do; I can't be the first person to run into this.

**Reply by bakerkretzmar** [reply]
You shouldn't need a `ziggy.js` file to run `npm install`, do you get some kind of error if you try to do that?

**Reply by plweil** [reply]
Sorry, I should have written that more carefully. We have `npm run production` as a postinstall script. So it's really a webpack error.

**Reply by bakerkretzmar** [reply]
Gotcha. Yeah I think you should just commit the whole `ziggy.js` file to the repo, way easier that way and I don't think it's 'wrong'.
